compiler/AST/build.cpp

```diff
@@ -1736,6 +1719,14 @@ buildClassDefExpr(const char* name,
                   Flag        isExtern,
                   const char* docs) {
   AggregateType* ct = toAggregateType(type);
+  if (strcmp("string", name) == 0) {
```

Comment:
I believe we have best practice not to strcmp in the compiler. Could you instead check that we're talking about the record marked with "string record" ? Is the issue that we're too early in the process to use the usual thing (where dtWhatever types are identified in types.cpp)?